src/analyzer/Analyzer.cc

```diff
@@ -117,6 +117,8 @@ Analyzer::~Analyzer() {
     // Make sure any late entries into the analyzer tree are handled (e.g.
     // from some Done() implementation).
     LOOP_OVER_GIVEN_CHILDREN(i, new_children) {
+        reporter->InternalWarning("%s has new child %s in analyzer destructor", fmt_analyzer(this).c_str(),
+                                  fmt_analyzer(*i).c_str());
```

Comment:
Drop the whole block ,replace with `assert(new_children.empty())` after testing in production.